### PR TITLE
Make case classes final

### PR DESCRIPTION
**Pull request by mdedetrich**
Its idiomatic for case classes to be final since users overriding the case class can cause unexpected behaviour

### CODE REVIEW

File: src/main/scala/com/github/sbt/sbom/BomExtractorParams.scala [hunk 1]
```diff
@@ -3,4 +3,4 @@ package com.github.sbt.sbom
 import org.cyclonedx.Version
 import sbt.Configuration
 
-case class BomExtractorParams(schemaVersion: Version, configuration: Configuration)
+final case class BomExtractorParams(schemaVersion: Version, configuration: Configuration)
```
Comment (eed3si9n) [inline src/main/scala/com/github/sbt/sbom/BomExtractorParams.scala:6]: Idiomatically all case classes are assumed to be final. Any attempt to extend them is non-idiomatic, and asking for trouble. For sbt plugins I don't think putting this guardrail is not necessary.

**Review by raboof — APPROVED** [review]
I don't think it is universally 'idiomatic' to explicitly mark case classes as final, and I personally find that in this case it adds noise for questionable benefit, but if you feel strongly I'm OK with it.

**Comment by mdedetrich** [comment]
> I don't think it is universally 'idiomatic' to explicitly mark case classes as final, and I personally find that in this case it adds noise for questionable benefit, but if you feel strongly I'm OK with it.

Yeah tbh this is something that should be solved by a linting tool, there was a big discussion somewhere that `case class` should be `final` by default and the only argument against it was some niche Scala2 scala usecase that was given by Martin.

So you are right that its not idiomatic to mark `case class` as final (although a lot of "serious" projects in open source Scala do this) but treating `case class` as final is almost universally idiomatic for reasons given before.

**Comment by raboof** [comment]
> treating `case class` as final is almost universally idiomatic

Yes, no disagreement there :)

**Pull request merged**